services/src/main/java/org/keycloak/protocol/oidc/grants/AuthorizationCodeGrantType.java

```diff
@@ -127,24 +130,32 @@ public Response process(Context context) {
         }
 
         if (redirectUri != null && !redirectUri.equals(redirectUriParam)) {
-            event.error(Errors.INVALID_CODE);
-            logger.tracef("Parameter 'redirect_uri' did not match originally saved redirect URI used in initial OIDC request. Saved redirectUri: %s, redirectUri parameter: %s", redirectUri, redirectUriParam);
+            event.error(Errors.INVALID_REDIRECT_URI);
+            String errorMessage = "Parameter 'redirect_uri' did not match originally saved redirect URI used in initial OIDC request. Saved redirectUri: %s, redirectUri parameter: %s";
+            event.detail(Details.REASON, String.format(errorMessage, redirectUri, redirectUriParam));
+            logger.tracef(errorMessage, redirectUri, redirectUriParam);
             throw new CorsErrorResponseException(cors, OAuthErrorException.INVALID_GRANT, "Incorrect redirect_uri", Response.Status.BAD_REQUEST);
         }
 
         if (!client.getClientId().equals(clientSession.getClient().getClientId())) {
-            event.error(Errors.INVALID_CODE);
-            throw new CorsErrorResponseException(cors, OAuthErrorException.INVALID_GRANT, "Auth error", Response.Status.BAD_REQUEST);
+            event.error(Errors.INVALID_CLIENT);
```

Comment:
Here I used the more appropriate `Errors.INVALID_CLIENT` instead of `Errors.INVALID_CODE`